### PR TITLE
Disable SqlServerDatabaseCreatorTest on CI

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Refactor SqlServerDatabaseCreatorTest

Part of #8615

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs [hunk 1]
```diff
@@ -4,449 +4,464 @@
...
 using Microsoft.EntityFrameworkCore.Storage.Internal;
+using Microsoft.EntityFrameworkCore.TestUtilities.Xunit;
+using Microsoft.EntityFrameworkCore.Utilities;
 using Microsoft.Extensions.DependencyInjection;
 using Xunit;
 
 namespace Microsoft.EntityFrameworkCore
 {
-    public class SqlServerDatabaseCreatorExistsTest
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
```
Comment (anpete) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: When did this become a thing? For consistency use ```// ReSharper disable``` syntax

File: test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs [hunk 2]
```diff
@@ -4,449 +4,464 @@
...
+    public class SqlServerDatabaseCreatorTest
     {
-        [Fact]
-        public async Task Exists_returns_false_when_database_doesnt_exist()
+        public class Exists
         {
-            await Exists_returns_false_when_database_doesnt_exist_test(async: false, file: false);
-        }
+            [ConditionalFact]
+            [SqlServerCondition(SqlServerCondition.IsNotTeamCity)]
```
Comment (anpete) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: Curious, can these just be put on the class itself? (DRY)

Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: Even if we could they have different flags on different test

Comment (anpete) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: XUnit could combine :-)

Comment (anpete) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: Or we could

Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: That would add complexity for little value

Comment (anpete) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: Disagree. Elegant and DRY.

Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: `IsNoTeamCity` on class.
and all tests would have `ConditionalFact`

Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/SqlServerDatabaseCreatorTest.cs]: Ok, it works if I put it on the immediate class, so it's not completely DRY, but better

**Pull request merged**